stdlib/tests/mem/mod.rs

```diff
@@ -0,0 +1,129 @@
+use test_utils::{
```

Comment:
I had to rename `mem.rs` to `mem/mod.rs` because the former was being compiled twice, once as part of the `tests/main.rs` and another time because cargo compiles all the integration tests in the `tests` folder. The issue is that the later didn't have the `build_test` macro (also, I think the tests are executed twice)

Reply:
I'm curious what's the idiomatic way to handle this. @vlopes11 - maybe you can recommend?

Reply:
That is the way to go. `cargo test` will compile anything inside `tests`, and execute the functions that contains the directive `#[test]`. If we want to create a utilities library for tests, we should `tests/foo/mod.rs` - as in the introduced change.

If we happen to contain no `#[test]` functions, cargo will compile it anyway.